python/lsst/ts/mtaos/model.py

```diff
@@ -1021,6 +1026,85 @@ def _generate_pipetask_command(
 
         return run_pipetask_cmd
 
+    async def get_image_info(self, visit_id):
+        """Get image information from the butler.
+
+        Parameters
+        ----------
+        visit_id : `int`
+            Visit id of the image.
+
+        Returns
+        -------
+        filter : `str`
+            Filter of the image.
+        rotation_angle : `float`
+            Rotation angle of the image.
+        elevation : `float`
+            Elevation of the image.
+
+        Raises
+        ------
+        ValueError
+            If the visit id is not found in the butler.
+        """
+        butler = dafButler.Butler(self.data_path)
+        refs = butler.query_datasets("raw", where=f"visit={visit_id}")
+
+        if len(refs) == 0:
+            raise ValueError(
+                f"Visit {visit_id} has no associated images in the butler."
+            )
+
+        image = butler.get(refs[0])
+        filter = image.getFilter().bandLabel
+        rotation_angle = image.getMetadata().get("ROTPA")
+        elevation = (
+            image.getMetadata().get("ELSTART") + image.getMetadata().get("ELEND")
+        ) / 2
+
+        return filter, rotation_angle, elevation
+
+    async def should_apply_corrections(self, visit_id: int) -> float:
+        """Check if corrections are supposed to be applied.
+
+        Parameters
+        ----------
+        visit_id : `int`
+            Visit id of the image.
+
+        Returns
+        -------
+        gain : `float`
+            Gain to apply to the corrections.
+        """
+        _, _, elevation = await self.get_image_info(visit_id)
+        _, _, prev_elevation = await self.get_image_info(visit_id - 1)
```

Comment:
This is a bit dangerous, you can't really say that `visit_id - 1`  is a good reference point. 


Instead, you might want to keep track of the elevation of the data that was last processed/applied and use it instead. For the first one of the sequence, you could just ignore the check.